mqtrigger/messageQueue/messageQueue.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2016 The Fission Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package messageQueue
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/fission/fission"
+	controllerClient "github.com/fission/fission/controller/client"
+)
+
+type MessageQueueConfig struct {
+	MQType string
+	Url    string
+}
+
+const (
+	NATS string = "nats-streaming"
+)
+
+type (
+	MessageQueue interface {
```

Comment:
Suggestion -- no need for `triggerSubscription` in this interface. Just return an `interface{}` from `subscribe` and pass the same to `unsubscribe`. The implementations wouldn't use any field other than `triggerSubscription.subscription` anyway.